### PR TITLE
[REF] Dockerfile template: If exists validation

### PR DESCRIPTION
**Pull request by moylop260**

### CODE REVIEW

**Comment by moylop260** [comment]
@lasley 
FYI here is my proposal to fix the use of original image of travis, another random image and vauxoo image.

File: src/travis2docker/templates/Dockerfile [hunk 1]
```diff
@@ -58,9 +58,9 @@ WORKDIR ${TRAVIS_BUILD_DIR}
 
 {% if runs -%}
 {% if image == 'quay.io/travisci/travis-python' -%}
-RUN /bin/bash -c "source $HOME/virtualenv/python2.7_with_system_site_packages/bin/activate && {{ ' && '.join(runs) }}"
+RUN /bin/bash -c "source $HOME/virtualenv/python2.7_with_system_site_packages/bin/activate && source /home/travis/.rvm/scripts/rvm  && {{ ' && '.join(runs) }}"
 {% else %}
-RUN /bin/bash -c "source /usr/local/rvm/scripts/rvm && {{ ' && '.join(runs) }}"
+RUN /bin/bash -c "test -f /usr/local/rvm/scripts/rvm && source /usr/local/rvm/scripts/rvm; {{ ' && '.join(runs) }}"
```
Comment (lasley) [inline src/travis2docker/templates/Dockerfile:63]: This won't work if RVM is installed under the user on the unknown image. RVM recommends that we do the following, which covers the Travis, the known, and the unknown:

```
# Load RVM into a shell session *as a function*
if [[ -s "$HOME/.rvm/scripts/rvm" ]] ; then

  # First try to load from a user install
  source "$HOME/.rvm/scripts/rvm"

elif [[ -s "/usr/local/rvm/scripts/rvm" ]] ; then

  # Then try to load from a root install
  source "/usr/local/rvm/scripts/rvm"

else

  printf "ERROR: An RVM installation was not found.\n"

fi
```

https://rvm.io/workflow/scripting

I was just going to make a shell file, drop the file on the container with `ADD`, then just run it before the run commands. Would reduce the code duplication too, letting us change it in the future easier if necessary.

Comment (moylop260) [reply src/travis2docker/templates/Dockerfile:63]: I got it
Thanks +1

**Comment by lasley** [comment]
Thanks @moylop260 - closing in favor of https://github.com/Vauxoo/travis2docker/pull/77/commits/caa9769cd628c45f45ecb753f33f7bd981773ed2

**Pull request closed without merging**